src/with_ladj.jl

```diff
@@ -76,17 +76,30 @@ export with_logabsdet_jacobian
 end
 
 
-@inline _get_y(y_with_ladj::NTuple{2,Any,}) = y_with_ladj[1]
-@inline _get_ladj(y_with_ladj::NTuple{2,Any}) = y_with_ladj[2]
-
-_with_ladj_on_mapped(map_or_bc::Function, y_with_ladj::Tuple{Any,Real}) = y_with_ladj
+function _with_ladj_on_mapped(map_or_bc::F, y_with_ladj::Tuple{Any,Real})  where {F<:Union{typeof(map),typeof(broadcast)}}
+    return y_with_ladj
+end
 
-function _with_ladj_on_mapped(map_or_bc::Function, y_with_ladj)
-    y = map_or_bc(_get_y, y_with_ladj)
-    ladj = sum(map_or_bc(_get_ladj, y_with_ladj))
+function _with_ladj_on_mapped(map_or_bc::F, y_with_ladj) where {F<:Union{typeof(map),typeof(broadcast)}}
+    y = map_or_bc(first, y_with_ladj)
+    ladj = sum(Broadcast.instantiate(Broadcast.broadcasted(last, y_with_ladj)))
     (y, ladj)
 end
 
+
+# Need to use a type for this, type inference fails when using a pullback
+# closure over YLT in the rrule, resulting in bad performance:
+struct WithLadjOnMappedPullback{YLT} <: Function end
+function (::WithLadjOnMappedPullback{YLT})(thunked_ΔΩ) where YLT
+    ys, ladj = unthunk(thunked_ΔΩ)
+    return NoTangent(), NoTangent(), broadcast((y, l) -> Tangent{YLT}(y, l), ys, ladj)
```

Comment:
It seems this is just
```suggestion
    return NoTangent(), NoTangent(), map(y -> Tangent{YLT}(y, ladj), ys)
```
so one could avoid broadcasting?

Reply:
With broadcasting, we could take advantage of user specializations, though (well, if the only specialize broadcasting, not `map`). What do you think?

Reply:
True

Reply:
No difference in benchmark and code it simpler, let's take your variant.